src/ArgumentResolver/VersionValueResolver.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt\Docs\ArgumentResolver;
+
+use Bolt\Docs\Documentation;
+use Bolt\Docs\Version;
+use Generator;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+/**
+ * Version value resolver.
+ *
+ * @author Gawain Lynch <gawain.lynch@gmail.com>
+ */
+final class VersionValueResolver implements ArgumentValueResolverInterface
```

Comment:
This is a lot of code just to skip a getter...WDYT?

A `Documentation::getVersionOrDefault` could remove the conditional logic in each place.

Reply:
13 LLoC … :wink: 

But I'll cover the reasoning below

Reply:
> also, this is essentially what was being done previously with the ->convert()

Yeah I know. My initial hope when coding was that `Documentation` class wouldn't be needed in the controller methods and the selected `Version` would be enough. But it turned out not to be the case.